lib/constructs/fetch-cross-stack-values.ts

```diff
@@ -0,0 +1,242 @@
+        ),
+        tableStreamArn: StringParameter.valueForStringParameter(
+          this,
+          name(buildConfig, "permissionSetTableStreamArn")
+        ),
+      }
+    );
+
+    this.permissionSetArnTable = Table.fromTableArn(
+      this,
+      name(buildConfig, "importedPermissionSetArnTable"),
+      StringParameter.valueForStringParameter(
+        this,
+        name(buildConfig, "permissionSetArnTableArn")
+      )
+    );
+
+    this.snsTopicsKey = Key.fromKeyArn(
+      this,
+      name(buildConfig, "importedSnsTopicsKey"),
+      StringParameter.valueForStringParameter(
+        this,
+        name(buildConfig, "snsTopicsKeyArn")
+      )
+    );
+    this.ddbTablesKey = Key.fromKeyArn(
+      this,
+      name(buildConfig, "importedDdbTablesKey"),
+      StringParameter.valueForStringParameter(
+        this,
+        name(buildConfig, "ddbTablesKeyArn")
+      )
+    );
+
+    this.permissionSetHandlerSSOAPIRoleArn = new SSMParamReader(
+      this,
+      name(buildConfig, "permissionSetHandlerSSOAPIRoleArnpr"),
+      buildConfig,
+      {
+        ParamAccountId: buildConfig.PipelineSettings.SSOServiceAccountId,
+        ParamRegion: buildConfig.PipelineSettings.SSOServiceAccountRegion,
+        ParamNameKey: "permissionSetHandler-ssoapi-roleArn",
+        LambdaLayers: this.nodeJsLayer,
+      }
+    ).paramValue;
+
+    this.linkManagerHandlerSSOAPIRoleArn = new SSMParamReader(
+      this,
+      name(buildConfig, "linkManagerHandlerSSOAPIRoleArnpr"),
+      buildConfig,
+      {
+        ParamAccountId: buildConfig.PipelineSettings.SSOServiceAccountId,
+        ParamRegion: buildConfig.PipelineSettings.SSOServiceAccountRegion,
+        ParamNameKey: "linkManagerHandler-ssoapi-roleArn",
+        LambdaLayers: this.nodeJsLayer,
+      }
+    ).paramValue;
+
+    this.listInstancesSSOAPIRoleArn = new SSMParamReader(
+      this,
+      name(buildConfig, "listInstancesSSOAPIRoleArnpr"),
+      buildConfig,
+      {
+        ParamAccountId: buildConfig.PipelineSettings.SSOServiceAccountId,
+        ParamRegion: buildConfig.PipelineSettings.SSOServiceAccountRegion,
+        ParamNameKey: "listInstances-ssoapi-roleArn",
+        LambdaLayers: this.nodeJsLayer,
+      }
+    ).paramValue;
+
+    this.listGroupsIdentityStoreAPIRoleArn = new SSMParamReader(
+      this,
+      name(buildConfig, "listGroupsIdentityStoreAPIRoleArnpr"),
+      buildConfig,
+      {
+        ParamAccountId: buildConfig.PipelineSettings.SSOServiceAccountId,
+        ParamRegion: buildConfig.PipelineSettings.SSOServiceAccountRegion,
+        ParamNameKey: "listgroups-identitystoreapi-roleArn",
+        LambdaLayers: this.nodeJsLayer,
+      }
+    ).paramValue;
+
+    this.processTargetAccountSMInvokeRoleArn = new SSMParamReader(
+      this,
+      name(buildConfig, "processTargetAccountSMInvokeRoleArnpr"),
+      buildConfig,
+      {
+        ParamAccountId: buildConfig.PipelineSettings.OrgMainAccountId,
+        ParamRegion: "us-east-1",
```

Comment:
Would be nice to start commenting as we go, like explaining why this has to be hard set to us-east-1